sources/web/datalab/polymer/notebook.ts

```diff
@@ -44,7 +44,13 @@ async function processMessageEvent(e: MessageEvent) {
   const message = e.data as IframeMessage;
 
   if (message.command === CommandId.UPLOAD_USER_CREDS) {
-    ApiManagerFactory.getInstance().uploadOauthAccessToken();
+    try {
+      await ApiManagerFactory.getInstance().uploadOauthAccessToken();
+    } catch (e) {
+      message.command = CommandId.ERROR;
+      message.arguments = e;
+    }
+    sendMessageToNotebookEditor(message);
```

Comment:
Should this be inside the catch?

Reply:
We want to send an ack always, the catch is just changing the command to error and sending the error along.

Reply:
You are essentially reusing 'message' for a completely different purpose. I think the intent of the code would be more clear if you did this:
try {
  await ApiManagerFactory...
  sendMessageToNotebookEditor(message);
} catch (e) {
  const errorMessage = {
    command: CommandId.ERROR;
    arguments: e
  };
  sendMessageToNotebookEditor(errorMessage);
  return;
}

Reply:
Done. PTAL.